core/app/models/spree/payment.rb

```diff
@@ -17,6 +17,14 @@ class Payment < Spree::Base
     belongs_to :payment_method, -> { with_discarded }, class_name: 'Spree::PaymentMethod', inverse_of: :payments, optional: true
 
     has_many :offsets, -> { offset_payment }, class_name: "Spree::Payment", foreign_key: :source_id
```

Comment:
What about also deprecating the `offset_payment` scope?

Reply:
Good call. I deprecated `.offset_payment`, which is in turn use by the `.offsets` association.